src/employees.service.ts

```diff
@@ -0,0 +1,147 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, of, throwError } from 'rxjs';
+import { delay } from 'rxjs/operators';
+import * as employeeHier from './employees.json';
+
+const CREATE_ACTION = 'create';
+const UPDATE_ACTION = 'update';
+
+const makeFlatData = (data, idField, parentField, containsField) => {
+    const result = [];
+    function visitChildren(childrenArray: any[], parentId: any): void {
+        if (!childrenArray ) {
+            return;
+        }
+        for (let obj of childrenArray) {
+            let blank = {};
+            blank[parentField] = parentId;
+            result.push(Object.assign(blank, obj));
+            visitChildren(obj[containsField], obj[idField]);
+        }
+    }
+    visitChildren([data[0]], null);
```

Comment:
Consider iterating the entire data in case we want to support multiple root-level items.